src/UCTSearch.cpp

```diff
@@ -309,7 +309,7 @@ size_t UCTSearch::prune_noncontenders() {
         Nfirst = std::max(Nfirst, node->get_visits());
     }
     const auto min_required_visits =
-        Nfirst - est_playouts_left();
+        Nfirst - (int)(0.9*est_playouts_left());
```

Comment:
This seems the opposite direction from what I would expect.  I've generally seen nps curves increase over the duration of the move, which would suggest that est_playouts_left is an underestimate early on in the move, so multiplying it by a constant less than 1 would seem to amplify that effect.

In any case I would suggest this is not the place for this logic - it should be in est_playouts_left.  Since it probably technically only belongs in one branch of the if/else if/else.
(and I wonder why changing the optimumTime multiplier to 1.4 and removing this multiplier, is not as effective.)

Reply:
This might be why it works so well on my cpu, but not on my gpu...

Reply:
I think this is the right place for what I intended. If est_playouts is incurrate that should be fixed in that method, not here. But if you expect not all of the playouts to land on the 2nd best move, or to be spread out between 2nd and 3rd, or 1st and 2nd... then this is the place to account for that.

Reply:
ahh okay.  Still I don't think that kind of adjustment makes sense for the nodes/playouts limit case, or the 'too early to estimate velocity' case - and this 'first order constant' approach is identical to adjusting optimumTime in practice.  I think you really want a more fancy model if you are going to adjust here.

Reply:
I was hoping that this would get merged as a simple fix that clearly gains over stock, and then someone else will do something more complicated once they see it's probably easy to improve this a bit with some math based on puct and priors. SF uses improvingFactor scaler kinda like this. Using extra time if the 2nd is looking like it will overtake with more playouts would also be a win, but probably not be right here.

Reply:
My counter point is - dropping this part of the PR and changing the multiplier for 1.5 to 1.4 (or 1.35) - should have identical results and only makes it even more likely for this PR to get merged.

Reply:
Actually I take that back, I missevaluated how this affects things - it might not be close enough to the same as reducing the multiplier.

Reply:
This will prune earlier when the 2nd best choice is not about to catch up. We only want to spend more time when it is likely the 2nd choice will catch up, not whenever it is theoretically possible...

Reply:
Why don't you test this PR vs using 1.1 scaler here instead of 0.9 since you think it est_playouts is underestimated?